### DISCUSSION TITLE
Feedback on the debugging and refactoring work to introduce new Visitor Design Pattern

### DISCUSSION THREAD

**Discussion by tommens** [LemaireEmilien/calculator-cucumber-2024#14] [Ideas]
You have introduced a new Visitor Design Pattern to deal with the problem of having the pre/in/postfix notation hardcoded in the Expression class hierarchy. However, you did not fix the related problem that the counting methods (countOps, countNbs, countDepth) are still part of the Expression class hierarchy. Ideally, these would also need to be moved into a Visitor design pattern. (Each type of counting would correspond to a separate visitor.) 
I opened a new Issue #15 for this.

**Comment by MokonaNico** [comment]
This has been fixed in the pull request #17
